core/controllers/practice_sessions.py

```diff
@@ -39,15 +48,29 @@ class PracticeSessionsPageDataHandler(base.BaseHandler):
     """Fetches relevant data for the practice sessions page."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'classroom_url_fragment': constants.SCHEMA_FOR_CLASSROOM_URL_FRAGMENTS,
+        'topic_url_fragment': constants.SCHEMA_FOR_TOPIC_URL_FRAGMENTS
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'selected_subtopic_ids': {
+                'schema': {
+                    'type': 'custom',
+                    'obj_type': 'ConvertCommaSepearatedToProperList'
```

Comment:
Btw, can we unify this approach with the approach that we discussed in your other PR where there are `stringifiend_…` variables?

Reply:
I was also thinking the same thing initially because in that PR I have also used a cutom class to validate and normalize ```stringfied_exploration_ids```, but the problem is that -- we have to segregate the class based on some logic so that in one case we can use ```json.loads``` to modify stringified_exploration_ids and in other case we can split based on comma seperated values.

The problem is we cannot introduce any external other parameter into the custom classes instead of raw object. I have looked into the codebase and also tried to some extent.
In my view -- handling them differently will make them clear & function oriented.

Looking to your suggestions.
Thank you.

Reply:
But I mean can we modify the frontend to send the data in the same format and then have one common object for both?

Reply:
Yes we can do that.
 I have updated comma_seperated_ids into stringified list of ids please take a look.